packages/components/src/components/divider/divider.component.ts

```diff
@@ -0,0 +1,297 @@
+   * @default 'negative'
+   */
+  @property({ type: String, attribute: 'arrow-direction', reflect: true })
+  arrowDirection: string = DEFAULTS.ARROW_DIRECTION;
+
+  /**
+   * Position of the button, if applicable.
+   * - **positive**
+   * - **negative**
+   *
+   * Note: Positive and Negative directions are defined based on Cartesian plane.
+   * @default 'negative'
+   */
+  @property({ type: String, attribute: 'button-position', reflect: true })
+  buttonPosition: string = DEFAULTS.BUTTON_DIRECTION;
+
+  /** @internal */
+  @state()
+  private dividerTypeInternal: DividerTypeInternal = DEFAULTS.TYPE_INTERNAL;
+
+  /** @internal */
+  @state()
+  private observer: MutationObserver;
+
+  /** @internal */
+  @state()
+  private windowDirection: string = 'ltr';
+
+  constructor() {
+    super();
+    this.observer = new MutationObserver(() => this.handleWindowDirectionChange());
+  }
+
+  private handleWindowDirectionChange() {
+    const currentDirection = getComputedStyle(this).direction;
+
+    if (this.windowDirection !== currentDirection) {
+      this.windowDirection = currentDirection;
+    }
+    this.setGrabberButton();
+  }
+
+  override connectedCallback() {
+    super.connectedCallback();
+    this.observer.observe(document.documentElement, {
+      attributes: true,
+      attributeFilter: ['dir'],
+    });
+  }
+
+  override disconnectedCallback() {
+    super.disconnectedCallback();
+    this.observer.disconnect();
+  }
+
+  /**
+   * Sets the variant attribute for the divider component.
+   * If the provided variant is not included in the DIVIDER_VARIANT,
+   * it defaults to the value specified in DEFAULTS.VARIANT.
+   *
+   * @param variant - The variant to set.
+   */
+  private setVariant(variant: DividerVariant) {
+    this.setAttribute('variant', Object.values(DIVIDER_VARIANT).includes(variant) ? variant : DEFAULTS.VARIANT);
+  }
+
+  /**
+   * Sets the orientation attribute for the divider component.
+   * If the provided orientation is not included in the DIVIDER_ORIENTATION,
+   * it defaults to the value specified in DEFAULTS.ORIENTATION.
+   *
+   * @param orientation - The orientation to set.
+   */
+  private setOrientation(orientation: DividerOrientation) {
+    this.setAttribute(
+      'orientation',
+      Object.values(DIVIDER_ORIENTATION).includes(orientation) ? orientation : DEFAULTS.ORIENTATION,
+    );
+  }
+
+  /**
+   * Sets the buttonPosition and arrowDirection attribute for the divider component.
+   * If the provided buttonPosition and arrowDirection are not included in the DIRECTIONS,
+   * it defaults to the value specified in DIRECTIONS based on the ORIENTATION.
+   *
+   * @param buttonPosition - The buttonPosition to set.
+   * @param arrowDirection - The arrowDirection to set.
+   */
+  private ensureValidDirections() {
+    /* eslint-disable max-len */
+    const defaultDirection = this.orientation === DIVIDER_ORIENTATION.HORIZONTAL ? DIRECTIONS.NEGATIVE : DIRECTIONS.POSITIVE;
```

Comment:
```suggestion
    const defaultDirection = this.orientation === DIVIDER_ORIENTATION.HORIZONTAL
      ? DIRECTIONS.NEGATIVE
      : DIRECTIONS.POSITIVE;
```
By doing this, we can avoid using eslint-disabled.

Reply:
Done